pennylane/templates/embeddings/iqp.py

```diff
@@ -0,0 +1,273 @@
+
+        The ``features`` argument cannot be passed to the quantum node
+        as a positional argument. This is due to the fact that the embedding performs non-trivial calculations
+        on the features. As a consequence, the following code **will produce an error**:
+
+        .. code-block:: python
+
+            @qml.qnode(dev)
+            def circuit(features):
+               IQPEmbedding(features=features, wires=range(3), n_repeats=2)
+               return [qml.expval(qml.PauliZ(w)) for w in range(3)]
+
+            circuit([1., 2., 3.])
+
+        >>> ValueError: 'features' cannot be differentiable
+
+        **Repeating the embedding**
+
+        The embedding can be repeated by specifying the ``n_repeats`` argument:
+
+        .. code-block:: python
+
+            @qml.qnode(dev)
+            def circuit(features=None):
+                IQPEmbedding(features=features, wires=range(3), n_repeats=4)
+                return [qml.expval(qml.PauliZ(w)) for w in range(3)]
+
+            circuit(features=[1., 2., 3.])
+
+        Every repetition uses exactly the same quantum circuit.
+
+        **Using a custom entangler pattern**
+
+        A custom entangler pattern can be used by specifying the ``pattern`` argument. A pattern has to be
+        a nested list of dimension ``(K, 2)``, where ``K`` is the number of entanglers to apply.
+
+        .. code-block:: python
+
+            pattern = [[1, 2], [0, 2], [1, 0]]
+
+            @qml.qnode(dev)
+            def circuit(features=None):
+                IQPEmbedding(features=features, wires=range(3), pattern=pattern)
+                return [qml.expval(qml.PauliZ(w)) for w in range(3)]
+
+            circuit(features=[1., 2., 3.])
+
+        Since diagonal gates commute, the order of the wire pairs has no effect on the result.
+
+        .. code-block:: python
+
+            from pennylane import numpy as np
+
+            pattern1 = [[1, 2], [0, 2], [1, 0]]
+            pattern2 = [[1, 0], [0, 2], [1, 2]]  # a reshuffling of pattern1
+
+            @qml.qnode(dev)
+            def circuit(features=None, pattern=None):
+                IQPEmbedding(features=features, wires=range(3), pattern=pattern, n_repeats=3)
+                return [qml.expval(qml.PauliZ(w)) for w in range(3)]
+
+            res1 = circuit(features=[1., 2., 3.], pattern=pattern1)
+            res2 = circuit(features=[1., 2., 3.], pattern=pattern2)
+
+            assert np.allclose(res1, res2)
+
+        **Non-consecutive wires**
+
+        In principle, the user can also pass a non-consecutive wire list to the template.
+        For single qubit gates, the i'th feature is applied to the i'th wire index (which may not be the i'th wire).
+        For the entanglers, the product of i'th and j'th features is applied to the wire indices at the i'th and j'th
+        position in ``wires``.
+
+        For example, for ``wires=[2, 0, 1]`` the ``RZ` block applies the first feature to wire 2,
```

Comment:
```suggestion
        For example, for ``wires=[2, 0, 1]`` the ``RZ`` block applies the first feature to wire 2,
```

Reply:
Good catch, that was causing havoc in the docs